index.js

```diff
@@ -328,41 +342,46 @@ module.exports = {
         // prettier-ignore
         if (err) return cb(clarify(err, `Failed to get group details when adding members`))
 
-        const content = {
-          type: 'group/add-member',
-          version: 'v2',
-          secret: secret.toString('base64'),
-          root,
-          tangles: {
-            members: {
-              root,
-              previous: [root], // TODO calculate previous for members tangle
-            },
-            // likely incorrect group tangle and this will be overwritten by
-            // publish(), we just add it here to make the spec pass
-            group: {
-              root,
-              previous: [root],
+        getRootIdOfMsgAuthor(root, (err, rootAuthorId) => {
+          // prettier-ignore
+          if (err) return cb(clarify(err, "couldn't get root id of author of root msg"))
+
+          const content = {
+            type: 'group/add-member',
+            version: 'v2',
+            groupKey: secret.toString('base64'),
+            root,
+            creator: rootAuthorId,
```

Comment:
I might have missed this (`creator`) being added to the spec? I dislike that it's like author. Why do we need it?

> creator is the root metafeed id of the creator of the group 

Nope, I was involved! I think we should call it `groupCreatorRootFeedId` or `initAuthorRootrId` .. if it's not too late :D

Reply:
Not too late! But this PR just adds the checking against the spec. Changing the field name would IMO be a different issue/pr.